PrivateHospitals/backend/PrivateHospitals/PrivateHospitals.Application/Services/Appointment/AppointmentService.cs

```diff
@@ -22,80 +23,133 @@ public class AppointmentService(
 
-        if (speciality == null)
+        if (!isDoctorAvailable)
         {
-            return Result<List<AppointmentDto>>.ErrorResponse(new List<string> {"Doctor speciality can`t be null"});
+            return Result<bool>.ErrorResponse(new List<string> { "Doctor is not available" });
         }
 
-        var appointments = await _appointmentRepository.GetAppointmentBySpecialityAsync(patientId, speciality);
-        if (appointments.Count == 0)
+        if (appointmentDto.Date < DateTime.Now)
         {
-            return Result<List<AppointmentDto>>.ErrorResponse(new List<string> { "Appointment not found" });
+            return Result<bool>.ErrorResponse(new List<string> { "Appointment date has to be in the future" });
         }
-
-        var appointmentDto = _mapper.Map<List<AppointmentDto>>(appointments);
         
-        return Result<List<AppointmentDto>>.SuccessResponse(appointmentDto);
-    }
+        var appointment = _mapper.Map<Core.Models.Appointment>(appointmentDto);
+        appointment.AppointmentDate = appointmentDto.Date;
+        appointment.PatientId = patientId;
+        
+        await _appointmentRepository.CreateAppointmentAsync(appointment);
+        
+        return Result<bool>.SuccessResponse(true);
+     }
 
-    public async Task<Result<List<AppointmentDto>>> GetAppointmentsByDateAsync(string patientId, DateTime? fromDate, DateTime? toDate)
-    {
-        var patient = await _patientRepository.GetPatientByIdAsync(patientId);
-        if (patient == null)
-        {
-            return Result<List<AppointmentDto>>.ErrorResponse(new List<string> {"Patient not found"});
-        }
+     public async Task<Result<PaginationResult<AppointmentDto>>> GetAppointmentsAsync(AppointmentFilter appointmentFilter, string patientId)
+     {
+         if (appointmentFilter.Speciality != null)
+         {
+             if (appointmentFilter.PageNumber == null || appointmentFilter.PageSize == 0)
+             {
+                 return Result<PaginationResult<AppointmentDto>>.ErrorResponse(new List<string> { "Invalid page size or page number" });
+             }
 
-        if (fromDate == null || toDate == null)
-        {
-            return Result<List<AppointmentDto>>.ErrorResponse(new List<string> {"FromDate and ToDate cannot be null"});
-        }
+             var appointments = await _appointmentRepository.GetAppointmentBySpecialityAsync(appointmentFilter, patientId);
+             if (appointments.Items.Count == 0)
+             {
+                 return Result<PaginationResult<AppointmentDto>>.ErrorResponse(new List<string> { "No appointments found" });
+             }
+             
+             var appointmentsDto = _mapper.Map<PaginationResult<AppointmentDto>>(appointments);
+             
+             return Result<PaginationResult<AppointmentDto>>.SuccessResponse(appointmentsDto);
+         }
+         else if (appointmentFilter.ToDate.HasValue && appointmentFilter.FromDate.HasValue)
```

Comment:
Nope, these filters should be applied as `AND` in sql. Make something similar to it https://chatgpt.com/share/673f0858-6648-800b-90fb-85d9cbeb5be2